pinot-core/src/main/java/org/apache/pinot/core/transport/ServerInstance.java

```diff
@@ -173,4 +189,18 @@ public int hashCode() {
   public String toString() {
     return _instanceId;
   }
+
+  @VisibleForTesting
+  int extractGroup(InstanceConfig instanceConfig) {
+    Map<String, String> pools = instanceConfig.getRecord().getMapField(InstanceUtils.POOL_KEY);
```

Comment:
The type of the field `pools` of spi Instance class uses Map<String, Integer>. Thus it is safe to directly use Integer.parseInt without checking the parsing exception and fallback to the default value in case of parsing exception.